code/modules/research/xenobiology/xenobiology_dissections.dm

```diff
@@ -0,0 +1,1533 @@
+/obj/item/organ/internal/cell/xenobiology/supercharged/Initialize(mapload)
+	. = ..()
+	icon_state = pick("vetus1", "vetus2")
+
+/obj/item/organ/internal/cell/xenobiology/supercharged/insert(mob/living/carbon/human/M, special = 0, dont_remove_slot = 0)
+	. = ..()
+	var/datum/spell/charge_up/explode/spell = new
+	spell.quality = organ_quality
+	M.AddSpell(spell)
+
+/obj/item/organ/internal/cell/xenobiology/supercharged/remove(mob/living/carbon/human/M, special = 0)
+	. = ..()
+	M.RemoveSpell(/datum/spell/charge_up/explode)
+
+/datum/spell/charge_up/explode
+	name = "Detonate"
+	desc = "Build up energy in your core, allowing for a fiery detonation!"
+	clothes_req = FALSE
+	stat_allowed = CONSCIOUS
+	invocation_type = "none"
+	action_icon_state = "genetic_incendiary"
+	action_background_icon_state = "bg_flayer"
+	max_charge_time = 10 SECONDS
+	base_cooldown = 10 MINUTES
+	charge_sound = new /sound('sound/magic/lightning_chargeup.ogg', channel = 7)
+	stop_charging_text = "You cool your core before it goes critical."
+	start_charging_text = "You begin to overload your core!"
+	stop_charging_fail_text = "You try to stop your cores charge but it's too late!"
+	var/quality
+
+/datum/spell/charge_up/explode/New()
+	..()
+	charge_up_overlay = image(icon = 'icons/effects/effects.dmi', icon_state = "purplesparkles", layer = EFFECTS_LAYER)
+
+/datum/spell/charge_up/explode/create_new_targeting()
+	return new /datum/spell_targeting/self
+
+/datum/spell/charge_up/explode/Click()
+	if(cast_check(TRUE, FALSE, usr))
+		if(!start_time)
+			INVOKE_ASYNC(src, PROC_REF(StartChargeup), usr)
+		else
+			if(!try_stop_buildup(usr))
+				return // Don't remove the click intercept
+
+/datum/spell/charge_up/explode/Discharge(mob/user)
+	. = ..()
+	if(quality == ORGAN_DAMAGED)
+		explosion(user.loc, 1, 2, 3, 3, cause = user) // gaurenteed gib, just about
+	else
+		explosion(user.loc, 0, 2, 3, 3, cause = user)
+	if(quality == ORGAN_PRISTINE)
+		user.status_flags |= GODMODE
+		addtimer(CALLBACK(src, PROC_REF(cleanup), user), 0.3 SECONDS) // juuust long enough to survive the blast itself.
+
+/datum/spell/charge_up/explode/proc/cleanup(mob/living/carbon/human/user)
+	user.status_flags &= ~GODMODE
+	if(user.nutrition > 140) // overloading your powersource will do that
+		user.nutrition = 140
+	else
+		user.nutrition = max(user.nutrition - 75, 5)
+	for(var/obj/item/organ/internal/cell/C in user.internal_organs)
+		C.damage += 40 // ouch. Maybe dont blow up
+
+/obj/item/organ/internal/heart/xenobiology/megacarp
+	name = "Rancid Clump"
+	desc = "It reeks of fish and... soy sauce?"
+	analyzer_price = 40
+	can_paradox = TRUE
+
+/obj/item/organ/internal/heart/xenobiology/megacarp/insert(mob/living/carbon/human/M, special = 0, dont_remove_slot = 0)
+	if(!M.mind)
+		M.visible_message("The organ doesnt look like its going to fit right now! It refuses.")
+		return
+	var/datum/spell/shapeshift/megacarp/spell = new
+	spell.quality = organ_quality
+	M.mind.AddSpell(spell)
+	. = ..()
```

Comment:
Please convert these trailing `. = ..()`s to `return ..()`, I'm going to leave those up to you to catch for the rest of this file